### PR TITLE
Add `chwidget` annotation for `ConsumerHookWidget`

### PR DESCRIPTION
**Pull request by tim-smart**
This is using the `deps-update` branch as a base, to fix some issues around old dependencies.

### CODE REVIEW

File: packages/functional_widget/pubspec.yaml [hunk 1]
```diff
@@ -15,6 +15,9 @@ dependencies:
   source_gen: ^1.1.0
   code_builder: ^4.0.0
   collection: ^1.0.0
+    
+dependency_overrides:
```
Comment (rrousselGit) [inline packages/functional_widget/pubspec.yaml:19]: What is the purpose of this dependency override

Comment (tim-smart) [reply packages/functional_widget/pubspec.yaml:19]: Will need to upgrade `mockito` to get rid of it.

File: packages/functional_widget/test/src/success.dart [hunk 2]
```diff
@@ -73,6 +73,9 @@ Widget withPositionalOptional(int? foo) => Container();
 @hwidget
 Widget hookExample() => Container();
 
+@chwidget
+Widget consumerHookExample() => Container();
```
Comment (rrousselGit) [inline packages/functional_widget/test/src/success.dart:77]: For `ConsumerHookWidget`, we need to pass the `WidgetRef` to the widget

File: packages/functional_widget/lib/src/parameters.dart [hunk 3]
```diff
@@ -4,10 +4,24 @@ import 'package:analyzer/dart/element/nullability_suffix.dart';
...
 import 'package:build/build.dart';
 import 'package:code_builder/code_builder.dart';
+import 'package:collection/collection.dart';
 
 class FunctionParameters {
   FunctionParameters._(this._parameters);
 
+  static const nonUserDefinedTypeSymbols = [
+    'Key',
+    'Key?',
```
Comment (rrousselGit) [inline packages/functional_widget/lib/src/parameters.dart:14]: What is the purpose of allowing non-nullable key if later we do `key!` to pass it?

Comment (rrousselGit) [reply packages/functional_widget/lib/src/parameters.dart:14]: Ignore me, I didn't see that you added the `required Key key`

**Comment by rrousselGit** [comment]
LGTM thanks!

**Pull request merged**